osh/hub/scan/models.py

```diff
@@ -1283,3 +1283,31 @@ def csmock_args(self):
             # there are no arguments, this is not an error
             logger.info("No csmock arguments for profile '%s'", self)
             return ''
+
+
+class RetentionPolicySettingsManager(models.Manager):
+    def get_days_from_category(self, category):
+
+        # We establish a broad default days, so we don't delete unwanted results.
+        try:
+            days = self.get(category=category).days
+        except ObjectDoesNotExist:
+            logger.warning("Category %s could not be found. Setting default parameter", category)
+            days = 60
+
+        return days
+
+
+class RetentionPolicySetting(models.Model):
+    category = models.CharField(max_length=128, blank=False, null=False)
+    days = models.PositiveIntegerField(
+        default=14,
+        help_text="Number of days before the task is deleted."
+    )
+
+    objects = RetentionPolicySettingsManager()
+
+
+class RetentionScan(models.Model):
```

Comment:
What about something more specific such as `RetentionAppliedTask` ?

Reply:
I am not sure how much OpenScanHub is following Django's best practices but I think the name should start with `Task` because it is bound to the `Task` table, e.g. `TaskRemoval`, or `TaskPurge` (so that a plural of it makes sense).  There seem to be preceding cases with `Package -> PackageAttribute`, `Scan -> ScanBinding`, `Result -> ResultGroup`, etc.

I think we should also record a timestamp when the task data was removed so that we have the answer when users ask for it (and so that we can verify that the retention policy implementation really follow its configuration).

Reply:
Hey! I modified the names of this and created the `TaskResultsRemoval` model. Also, I have added a new field with the datetime when the task is added to the table